### ISSUE TITLE
Enum value in order by clause as int

### ISSUE THREAD

**Issue by akanmuratcimen** [aspnet/EntityFramework#6511]
I have simple model classes:

```
public class VehicleClass {
    public int Id { get; set; }
    public virtual ICollection<VehicleClassCulture> CulturedEntities { get; set; }
}

public class VehicleClassCulture{
    public int BaseEntityId { get; set; }
    public Culture Culture { get; set; }
    public string Name { get; set; }
    public Status Status { get; set; }
    public virtual VehicleClass BaseEntity { get; set; }
}
```

When i'm trying to run the query below:

```
from vehicleClass in _dbContext.VehicleClasses

join vehicleClassCulture in _dbContext.VehicleClassCultures on
new { vehicleClass.Id, Culture = CurrentCulture } equals
new { Id = vehicleClassCulture.BaseEntityId, vehicleClassCulture.Culture }
into currentCulture

join vehicleClassCulture in _dbContext.VehicleClassCultures on
new { vehicleClass.Id, Culture = DefaultCulture } equals
new { Id = vehicleClassCulture.BaseEntityId, vehicleClassCulture.Culture }
into fallbackCulture

from selected in currentCulture.DefaultIfEmpty()
from fallback in fallbackCulture.DefaultIfEmpty()
let culturedEntity = selected ?? fallback

select new {
    Id = vehicleClass.Id,
    Name = culturedEntity.Name,
    Status = culturedEntity.Status,
    Culture = culturedEntity.Culture,
};
```

The "CurrentCulture" and the "DefaultCulture" variables are enum types.

I'm getting this error: 

> "The SELECT item identified by the ORDER BY number 2 contains a variable as part of the expression identifying a column position. Variables are only allowed when ordering by an expression referencing a column name."

The generated query is: 

> declare @__CurrentCulture_0 int = 1 -- Manually added
> 
> SELECT [vehicleClass].[Id], [t].[Culture], [t].[VehicleClassId], [t].[Name], [t].[Status]  FROM [VehicleClass] AS [vehicleClass]  LEFT JOIN (      SELECT [x0].[Culture], [x0].[VehicleClassId], [x0].[Name], [x0].[Status]      FROM [VehicleClassCulture] AS [x0]  ) AS [t] ON ([vehicleClass].[Id] = [t].[VehicleClassId]) AND (@__CurrentCulture_0 = [t].[Culture]) ORDER BY [vehicleClass].[Id], **@__CurrentCulture_0**

**Comment by anpete** [comment]
@smitpatel Dupe?

**Comment by smitpatel** [comment]
dupe of #6580 fixed in #6656
